model/infinispan/src/main/java/org/keycloak/models/cache/infinispan/GroupAdapter.java

```diff
@@ -51,7 +51,7 @@ public GroupAdapter(CachedGroup cached, RealmCacheSession cacheSession, Keycloak
         this.cacheSession = cacheSession;
         this.keycloakSession = keycloakSession;
         this.realm = realm;
-        modelSupplier = this::getGroupModel;
+        modelSupplier = new LazyModel<>(this::getGroupModel);
```

Comment:
```suggestion
        modelSupplier = Suppliers.memoize(this::getGroupModel);
```

@pedroigor AFAIK, we might leverage the Guava Memoization with all the necessary checks, thread-safety, etc. as we already have the dependency there. WDYT?

We might consider using it also in other places with suppliers.

```
import com.google.common.base.Suppliers;
```